Slim/Control/Queries.pm

```diff
@@ -1882,6 +1882,147 @@ sub genresQuery {
 }
 
 
+sub rolesQuery {
+	my $request = shift;
+
+	# check this is the correct query.
+	if ($request->isNotQuery([['roles']])) {
+		$request->setStatusBadDispatch();
+		return;
+	}
+
+	if (!Slim::Schema::hasLibrary()) {
+		$request->setStatusNotDispatchable();
+		return;
+	}
+
+	my $sqllog = main::DEBUGLOG && logger('database.sql');
+
+	# get our parameters
+	my $client        = $request->client();
+	my $index         = $request->getParam('_index');
+	my $quantity      = $request->getParam('_quantity');
+	my $year          = $request->getParam('year');
+	my $contributorID = $request->getParam('artist_id');
+	my $albumID       = $request->getParam('album_id');
+	my $trackID       = $request->getParam('track_id');
+	my $workID        = $request->getParam('work_id');
+	my $libraryID     = Slim::Music::VirtualLibraries->getRealId($request->getParam('library_id'));
+	my $tags          = $request->getParam('tags') || '';
```

Comment:
Tags is unused. But should there be support for eg. requesting the number of roles?

Reply:
Another tag might be to return not only the numerical ID, but the textual as well. I could imagine that under certain circumstances this would be helpful.

Reply:
Makes sense.